config/config_helpers.go

```diff
@@ -144,7 +164,7 @@ func (context *resolveContext) getHelperFunctionsHCLContext() (*hcl.EvalContext,
 					}
 					return cty.ListVal(outVals), nil
 				}
-				return cty.NullVal(helperFunction.returnType), fmt.Errorf("unsupported return type to %s. Type: %s", key, returnType)
+				panic(fmt.Errorf("unsupported return type to %s. Type: %s", key, returnType))
```

Comment:
Is it OK to panic here?  That's going to abruptly kill the process unless it's being recovered somewhere else.  Shouldn't we return an error instead, since it's a Go best practice to panic only on unrecoverable runtime errors? (https://yourbasic.org/golang/errors-explained/ and https://blog.golang.org/defer-panic-and-recover)

Reply:
This is an unrecoverable error since we do not manage returns value other that strings and list of strings.

But the panic call is handled right at the top of the same function and converted to error:
`defer errors.Recover(func(cause error) { err = cause })`.